File: java/ExtractXml.java
// Extract the raw factur-x.xml from a Factur-X PDF using the InvoiceXML API.
// Get API key: https://www.invoicexml.com/account/authentication
// Docs:        https://www.invoicexml.com/docs/api/extract/xml
//
// Dependency: com.squareup.okhttp3:okhttp:4.12.0

import okhttp3.*;
import java.io.File;
import java.nio.file.Files;
import java.nio.file.Paths;

public class ExtractXml {
    public static void main(String[] args) throws Exception {
        String apiKey = "YOUR_API_KEY";

        RequestBody body = new MultipartBody.Builder()
            .setType(MultipartBody.FORM)
            .addFormDataPart("file", "invoice.pdf",
                RequestBody.create(new File("invoice.pdf"), MediaType.parse("application/pdf")))
            .build();

        Request request = new Request.Builder()
            .url("https://api.invoicexml.com/v1/extract/xml")
            .header("Authorization", "Bearer " + apiKey)
            .post(body)
            .build();

        try (Response response = new OkHttpClient().newCall(request).execute()) {
            String xml = response.body().string();
            if (!response.isSuccessful()) {
                System.err.println("InvoiceXML API error " + response.code() + ": " + xml);
                System.exit(1);
            }
            Files.write(Paths.get("factur-x.xml"), xml.getBytes());
            System.out.println("Saved factur-x.xml (" + xml.length() + " chars)");
        }
    }
}

File: python/extract_xml.py
# Extract the raw factur-x.xml from a Factur-X PDF using the InvoiceXML API.
# Get API key: https://www.invoicexml.com/account/authentication
# Docs:        https://www.invoicexml.com/docs/api/extract/xml
#
# Dependency:  pip install requests

import requests
import sys

api_key = "YOUR_API_KEY"

response = requests.post(
    "https://api.invoicexml.com/v1/extract/xml",
    headers={"Authorization": f"Bearer {api_key}"},
    files={"file": ("invoice.pdf", open("invoice.pdf", "rb"), "application/pdf")},
)

if not response.ok:
    sys.stderr.write(f"InvoiceXML API error {response.status_code}: {response.text}\n")
    sys.exit(1)

with open("factur-x.xml", "w", encoding="utf-8") as f:
    f.write(response.text)

print(f"Saved factur-x.xml ({len(response.text)} chars)")
